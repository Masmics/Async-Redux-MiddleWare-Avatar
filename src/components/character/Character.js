import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function Character({ name, url }) {
  return (
    <Link to={`/${name}`}>
      <img src={url} />
      <h3>{name}</h3>
    </Link>
  );
}

Character.propTypes = {
  name: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};

export default Character;
