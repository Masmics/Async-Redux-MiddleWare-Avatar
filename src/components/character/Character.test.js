import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

describe('Character component', () => {
  it('renders a character', () => {
    const wrapper = shallow(<Character />);
    expect(wrapper).toMatchSnapshot();
  });
});
