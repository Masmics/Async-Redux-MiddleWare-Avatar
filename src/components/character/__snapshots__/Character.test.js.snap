// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders a character 1`] = `
<Link
  to="/undefined"
>
  <img />
  <h3 />
</Link>
`;
